File: ProjetPI(back)/src/main/java/tn/esprit/backend/entite/competences.java
package tn.esprit.backend.entite;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
public class competences {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long idCompetences;
    private String name;
    private String description;
    private int importanceLevel;

}

File: ProjetPI(front)/src/app/models/competences.ts
export class Competences {
  public idCompetences: number;
  public name: string;
  public description: string;
  public importanceLevel: number;
  public selected: boolean;

  constructor(
    idCompetences: number,
    name: string,
    description: string,
    importanceLevel: number,
    selected: boolean = false
  ) {
    this.idCompetences = idCompetences;
    this.name = name;
    this.description = description;
    this.importanceLevel = importanceLevel;
    this.selected = selected;
  }
}
